pkg/volume/gcepd/attacher.go

```diff
@@ -393,10 +402,42 @@ func (detacher *gcePersistentDiskDetacher) UnmountDevice(deviceMountPath string)
 		if err != nil {
 			return err
 		}
+		target, err := getPathTarget(deviceMountPath)
+		if err != nil {
+			return err
+		}
+		if err = dismount(target); err != nil {
+			return err
+		}
 	}
 	return mount.CleanupMountPoint(deviceMountPath, detacher.host.GetMounter(gcePersistentDiskPluginName), false)
 }
 
+// dismount the targeting device
+func dismount(target string) error {
+	cmd := fmt.Sprintf("mountvol %s /P", target)
+	output, err := exec.Command("cmd", "/c", cmd).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("dismount %s failed: %v, output: %q", target, err, string(output))
+	}
+	return nil
+}
+
+// Get device target from the device mount path
+func getPathTarget(path string) (string, error) {
+	cmd := fmt.Sprintf("(Get-Item %s).Target", path)
+	output, err := exec.Command("powershell", "/c", cmd).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("get path %s target failed: %v, output: %q", path, err, string(output))
+	}
+	target := string(output)
+	if len(target) < 1 {
+		return "", fmt.Errorf("get device path for %s failed, output is empty", target)
+	}
+	target = strings.Replace(target, "\r\n", "", -1)
```

Comment:
Would `strings.TrimSpace` or any of those variants work better?

Reply:
replaced.